<?xml version="1.0" encoding="UTF-8"?>
    <source package="java.lang.reflect">
        <class name="InvocationTargetException" line="28">
            <extends class="ReflectiveOperationException"/>
            <javadoc line="28">
                InvocationTargetException is a checked exception that wraps
                  an exception thrown by an invoked method or constructor.
                  &lt;p&gt;As of release 1.4, this exception has been retrofitted to conform to
                  the general purpose exception-chaining mechanism.  The &quot;target exception&quot;
                  that is provided at construction time and accessed via the{@link #getTargetException()} method is now known as the &lt;i&gt;cause&lt;/i&gt;,
                  and may be accessed via the {@link Throwable#getCause()} method,
                  as well as the aforementioned &quot;legacy method.&quot;                
                <see>
                    Method                    
                </see>
                <see>
                    Constructor                    
                </see>
            </javadoc>
            <declaration name="serialVersionUID" type="long" line="43"/>
            <javadoc line="43">
                Use serialVersionUID from JDK 1.1.X for interoperability                
            </javadoc>
            <declaration name="target" type="Throwable" line="48"/>
            <javadoc line="48">
                This field holds the target if the
                  InvocationTargetException(Throwable target) constructor was
                  used to instantiate the object                
                <serial>
                                        
                </serial>
            </javadoc>
            <javadoc line="58">
                Constructs an {@code InvocationTargetException} with{@code null} as the target exception.                
            </javadoc>
            <method name="InvocationTargetException" type="constructor" line="62">
                <comment line="63">
                    Disallow initCause                    
                </comment>
            </method>
            <javadoc line="66">
                Constructs a InvocationTargetException with a target exception.                
                <param>
                    target the target exception                    
                </param>
            </javadoc>
            <method name="InvocationTargetException" type="constructor" line="71">
                <params>
                    <param name="target" type="Throwable"/>
                </params>
                <comment line="72">
                    Disallow initCause                    
                </comment>
            </method>
            <javadoc line="76">
                Constructs a InvocationTargetException with a target exception
                  and a detail message.                
                <param>
                    target the target exception                    
                </param>
                <param>
                    s      the detail message                    
                </param>
            </javadoc>
            <method name="InvocationTargetException" type="constructor" line="83">
                <params>
                    <param name="target" type="Throwable"/>
                    <param name="s" type="String"/>
                </params>
                <comment line="84">
                    Disallow initCause                    
                </comment>
            </method>
            <javadoc line="88">
                Get the thrown target exception.
                  &lt;p&gt;This method predates the general-purpose exception chaining facility.
                  The {@link Throwable#getCause()} method is now the preferred means of
                  obtaining this information.                
                <return>
                    the thrown target exception (cause of this exception).                    
                </return>
            </javadoc>
            <method name="getTargetException" type="Throwable" line="97"/>
            <javadoc line="101">
                Returns the cause of this exception (the thrown target exception,
                  which may be {@code null}).                
                <return>
                    the cause of this exception.                    
                </return>
                <since>
                    1.4                    
                </since>
            </javadoc>
            <method name="getCause" type="Throwable" line="108"/>
        </class>
    </source>